### ISSUE TITLE
Trips up on sub JSON with query key

### ISSUE THREAD

**Issue by kmpatel** [wyolum/jsonlib#11]
```
const char json[] = R"({"parent":{"child":"div"},"child":"span"})";

jsonExtract(json,"child");     // should return "span", instead returns "div"
```

**Comment by wyojustin** [comment]
Thanks kmpatel for trying this lib and providing feedback!  Did you call  "jsonRemoveWhiteSpace()" first?

json = jsonRemoveWhiteSpace(json);

**Comment by kmpatel** [comment]
Oh.  No, I did not use jsonRemoveWhiteSpace. 
I saw it, but assumed it was a optional feature, not a mandatory step to parsing.

**Comment by wyojustin** [comment]
If I recall, will return first match for dups.

**Comment by kmpatel** [comment]
> If I recall, will return first match for dups.

I don't think you are paying attention to the example. I should have written "apparent duplicate" but, in the example, the call is to find the value of "child".  Instead, the fn returns the value of "parent.child", which is the wrong member.  It is a only a duplicate by appearance, when in fact it is a valid member of a sub-json "parent" that happens to appear before the target member "child".

**Comment by wyojustin** [comment]
Yes, I beleive you have identified a bug in the code.  The string searches are very rudimentary.  I'll look to see if there is an easy way to fix this, but you may want to swich to a more robust libary.
